libpod/runtime_pod_infra_linux.go

```diff
@@ -78,6 +78,11 @@ func (r *Runtime) createInfraContainer(ctx context.Context, p *Pod) (*Container,
 	}
 	imageName := newImage.Names()[0]
 	imageID := data.ID
+	entryCmd := []string{r.config.InfraCommand}
+	// default to entrypoint in image if there is one
+	if len(data.Config.Entrypoint) > 0 {
+		entryCmd = data.Config.Entrypoint
+	}
```

Comment:
While you're in here - can we also grab the image's environment variables and add them? Makes the infra container more of a standard container

Reply:
done, though the code is kind of clunky